concepts/05 Widgets/DataGrid/00 Getting Started with DataGrid/70 Configure the Master-Detail Interface.md

```diff
@@ -0,0 +1,276 @@
+The master-detail interface adds an expandable section that contains detail data (*"detail section"*) to a standard data row (*"master row"*). The detail section can contain anything.
+
+Master-detail data representation is configured in the [masterDetail](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/masterDetail/) object. To specify detail sections' contents, implement the [template](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/masterDetail/#template). Users can expand and collapse details sections if you set the [enabled](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/masterDetail/#enabled) option to **true**:
+
+---
+##### jQuery
+
+    <!-- tab: index.js -->
+    $(function() {
+        $("#dataGrid").dxDataGrid({
+            // ...
+            masterDetail: {
+                enabled: true,
+                template: function (_, options) {
+                    const employee = options.data;
+                    const photo = $("<img>")
+                        .addClass("employee-photo")
+                        .attr("src", employee.Photo);
+                    const notes = $("<p>")
+                        .addClass("employee-notes")
+                        .text(employee.Notes);
+                    return $("<div>").append(photo, notes);
+                }
+            },
+        });
+    });
+
+    <!-- tab: data.js -->
+    const employees = [{
+        "EmployeeID": 1,
+        "FullName": "Nancy Davolio",
+        "Position": "Sales Representative",
+        "TitleOfCourtesy": "Ms.",
+        "BirthDate": "1968-12-08T00:00:00.000Z",
+        "HireDate": "2011-05-01T00:00:00.000Z",
+        "Address": "507 - 20th Ave. E.\r\nApt. 2A",
+        "City": "Seattle",
+        "Region": "WA",
+        "PostalCode": "98122",
+        "Country": "USA",
+        "HomePhone": "(206) 555-9857",
+        "Extension": "5467",
+        "Photo": "https://js.devexpress.com/Demos/WidgetsGallery/JSDemos/images/employees/06.png",
+        "Notes": "Education includes a BA in psychology from Colorado State University in 1990.  She also completed \"The Art of the Cold Call.\"  Nancy is a member of Toastmasters International.",
+        "ReportsTo": 2
+    },
+    // ...
+    ];
+
+    <!-- tab: index.css -->
+    .employee-photo {
+        height: 140px;
+        float: left;
+        padding: 0 20px 20px 0;
+    }
+
+    .employee-notes {
+        text-align: justify;
+        white-space: normal;
+    }
+
+    /* ... */
+
+##### Angular
+
+    <!-- tab: app.component.html -->
+    <dx-data-grid ... >
+        <!-- ... -->
+        <dxo-master-detail
+            [enabled]="true"
+            [template]="'employee-info'">
```

Comment:
Is the second pair of quotes required?

I saw the following syntax in our docs: 
        <dxi-button template="myCommand">
            <div *dxTemplate="let data of 'myCommand'">
                <a href="#" (click)="logMyCommandClick()">My command</a>
            </div>
        </dxi-button>

Reply:
In this particular case, yes.
https://js.devexpress.com/Documentation/20_2/Guide/Angular_Components/Component_Configuration_Syntax/#Templates
> Angular has a built-in template directive. This causes an error when you try to specify an eponymous property on a configuration component (for instance, on dxo-master-detail).

Reply:
ok